lib/cublas/wrappers.jl

```diff
@@ -126,18 +126,29 @@ for (fname, fname_64, elty) in ((:cublasDscal_v2, :cublasDscal_v2_64, :Float64),
         end
     end
 end
+function scal!(n::Integer, alpha::Number, x::StridedCuVecOrDenseMat{T}) where {T}
+    gpu_α = CuRef{T}(alpha)
+    scal!(n, gpu_α, x)
+    synchronize()
+    return x
+end
+function scal!(n::Integer, alpha::CuRefArray{Float32, CuVector{Float32, M}}, x::StridedCuVecOrDenseMat{Float16}) where {M <: AbstractMemory}
+    cublasScalEx(handle(), n, alpha, Float32, x, Float16, stride(x, 1), Float32)
+    return x
+end
 function scal!(n::Integer, alpha::Number, x::StridedCuVecOrDenseMat{Float16})
-    α = convert(Float32, alpha)
-    cublasScalEx(handle(), n, Ref{Float32}(α), Float32, x, Float16, stride(x, 1), Float32)
+    α = CuRef{Float32}(alpha)
+    x = scal!(n, α, x)
+    synchronize()
     return x
 end
 # specific variants in case x is complex and alpha is real
 for (fname, fname_64, elty, celty) in ((:cublasCsscal_v2, :cublasCsscal_v2_64, :Float32, :ComplexF32),
                                        (:cublasZdscal_v2, :cublasZdscal_v2_64, :Float64, :ComplexF64))
     @eval begin
         function scal!(n::Integer,
-                       alpha::$elty,
-                       x::StridedCuVecOrDenseMat{$celty})
+                       alpha::CuRefArray{$elty, CuVector{$elty, M}},
```

Comment:
Any reason you're typing these so specifically? I"d just use `alpha::Ref{$elty}` (`CuRefArray` isn't `<: CuRef` because we can't have `CuRef` be abstract and work with `ccall`; `Ref` is special, see `jl_is_abstract_ref_type`).

Reply:
I simplified it, feel free to revert if there was a specific reason.